snapcraft/internal/errors.py

```diff
@@ -483,6 +484,29 @@ def __init__(self, path: str, extractor_name: str) -> None:
         super().__init__(path=path, extractor_name=extractor_name)
 
 
+class SnapcraftCommandError(SnapcraftError, CalledProcessError):
+    """Exception for generic command errors.
+
+    Processes should capture this error for specific messaging.
+    This exception carries the signature of CalledProcessError for backwards
+    compatibility.
+    """
+
+    fmt = (
+            'Failed to run {command!r}: '
+            'Exited with code {exit_code}.'
+    )
+
+    def __init__(self, *, command: str,
+                 call_error: CalledProcessError) -> None:
+        super().__init__(command=command, exit_code=call_error.returncode)
+        self.output = call_error.output
+        self.stdout = call_error.stdout
+        self.stderr = call_error.stderr
+        self.cmd = call_error.cmd
+        self.returncode = call_error.returncode
```

Comment:
Here is the __init__ from `CalledProcessError`, from which we inherit here:

```
def __init__(self, returncode, cmd, output=None, stderr=None):
    self.returncode = returncode
    self.cmd = cmd
    self.output = output
    self.stderr = stderr
```

I say just call that init properly and don't worry about setting those variables things ourselves.